### PR TITLE
build-aux: Use fallback-x11 instead of x11

### PR DESCRIPTION
**Pull request by Etaash-mathamsetty**
### Description
Fixes a warning when compiling/building the obs flatpak
fallback-x11 only enables x11 if the application doesn't use wayland, and instead falls back to x11. If the app uses wayland then the x11 socket is never made available I am assuming.

### Motivation and Context
removes a warning

### How Has This Been Tested?
Testing is unnecessary because it just uses the same behavior but removes a build time warning

### Types of changes
Bug fix

### Checklist:

- [x] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [x] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [x] My code is not on the master branch.
- [x] The code has been tested.
- [x] All commit messages are properly formatted and commits squashed where appropriate.
- [x] I have included updates to all appropriate documentation.

### CODE REVIEW

**Comment by gxalpha** [comment]
Please adjust the commit message according to our [commit guidelines](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst#commit-guidelines).
Please also expand on the description/motivation here. How is `fallback-x11` different from `x11` (and in particular, why is it the correct fix)?, etc.

**Comment by GeorgesStavracas** [comment]
The reason I specifically left `x11` and not `fallback-x11` is because some plugins may crash OBS if they can't connect to the X11 socket. One such example is the [Input Overlay plugin](https://github.com/flathub/com.obsproject.Studio.Plugin.InputOverlay).

Can you check if that's still the case?

**Comment by Etaash-mathamsetty** [comment]
> The reason I specifically left `x11` and not `fallback-x11` is because some plugins may crash OBS if they can't connect to the X11 socket. One such example is the [Input Overlay plugin](https://github.com/flathub/com.obsproject.Studio.Plugin.InputOverlay).
> 
> Can you check if that's still the case?

ah I didn't know about this

**Comment by GeorgesStavracas** [comment]
It would still be fantastic to replace `x11` with `fallback-x11` though, so it's worth checking and maybe even reporting the crash to the plugin authors. But I'm afraid of other plugins that could crash as well. Also, @gxalpha's comment still applies, a more comprehensive commit message would be great to have as well.

**Comment by Etaash-mathamsetty** [comment]
> It would still be fantastic to replace `x11` with `fallback-x11` though, so it's worth checking and maybe even reporting the crash to the plugin authors. But I'm afraid of other plugins that could crash as well. Also, @gxalpha's comment still applies, a more comprehensive commit message would be great to have as well.

Input overlay still uses x11, I am still wondering why they didn't use evdev though

**Comment by GeorgesStavracas** [comment]
I don't think it's possible to install polkit rules or elevate permissions from a Flatpak sandbox

**Comment by tytan652** [comment]
> Input overlay still uses x11, I am still wondering why they didn't use evdev though. It does need permissions, but polkit can handle that right?

No, the author/maintainers of the plugin will have to find (or contribute to) a new way to do it without punching holes in the sandbox and relying on the x11 socket to work on both X11 and Wayland.

If a plugin requires the x11 socket, it must be fixed to not make OBS crash under Wayland with no access to the x11 socket.

File: build-aux/com.obsproject.Studio.json
Comment (tytan652) [inline build-aux/com.obsproject.Studio.json:9]: Keep unresolved until the commit tittle is changed.

**Review by tytan652 — CHANGES_REQUESTED** [review]

**Comment by tytan652** [comment]
Input Overlay still crash with this kind of change.

**Comment by RytoEX** [comment]
cc @univrsal 
See above about an issue we're seeing when trying to clean up our CI warnings.
If I had to guess, the issue is with libuiohook, but I haven't actually checked.

**Comment by univrsal** [comment]
I'm pretty sure it's libuiohook. It could also be SLD2 but I assume that library behaves correctly. I can add a check and not start libuiohook if obs is running under wayland. 
```C
getenv("WAYLAND_DISPLAY") || obs_get_nix_platform() == OBS_NIX_PLATFORM_WAYLAND
```
Would either of these checks be enough?

**Comment by tytan652** [comment]
@univrsal

You should check if you are on X11/Xorg since it's the only supported "platform".
```
obs_get_nix_platform() != OBS_NIX_PLATFORM_X11_EGL
```
We dropped `_GLX` since a while so it can be ignored.

**Comment by univrsal** [comment]
I've added the check. Seems to work from what I can tell but I can't test if it fixes the crash as I don't have the flatpak build.

**Comment by RytoEX** [comment]
@GeorgesStavracas Is this something we want to try out and see what breakages occur in the wild? If so, when would you feel it's appropriate to do so?

**Comment by GeorgesStavracas** [comment]
I think we can try merging this right after the current release

**Comment by tytan652** [comment]
Blocked by #11019

File: build-aux/com.obsproject.Studio.json
Comment (RytoEX) [reply build-aux/com.obsproject.Studio.json:9]: I've fixed the commit message and PR title.

**Review by RytoEX — APPROVED** [review]
Approving based on prior conversation with @GeorgesStavracas .

**Pull request merged**
